File: cpp/test/TestSourceCharacter.cpp

#include "Source.h"

#include "gtest/gtest.h"


class SourceCharacterTest : public ::testing::Test {
protected:
    
};

TEST_F(SourceCharacterTest, Graphical1) {
    
    EXPECT_EQ(SourceCharacter('\t').graphicalString(), "\\t");
    
    EXPECT_EQ(SourceCharacter(0x1b).graphicalString(), "\\[RawEscape]");
    
    EXPECT_EQ(SourceCharacter(0xb0).graphicalString(), "\\[Degree]");
    
    EXPECT_EQ(SourceCharacter(0x80).graphicalString(), "\\.80");
    
    EXPECT_EQ(SourceCharacter(0xabcd).graphicalString(), "\\:abcd");
}

File: crates/wolfram-parser/src/tests/test_source_character.rs
use crate::read::WLCharacter;

#[test]
fn SourceCharacterTest_Graphical1() {
    assert_eq!(WLCharacter::new('\t').graphicalString(), "\\t");

    assert_eq!(WLCharacter::new(0x1b).graphicalString(), "\\[RawEscape]");

    assert_eq!(WLCharacter::new(0xb0).graphicalString(), "\\[Degree]");

    assert_eq!(WLCharacter::new('\u{abcd}').graphicalString(), "\\:abcd");
}
